### PR TITLE
TeX file for the pseudocode of the Longest Palindromic Substring problem

### PR DESCRIPTION
**Pull request by yildirimyigit**

### CODE REVIEW

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Find the longest palindromic sequence in a given string}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: Update the title to substring.

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  
+  \begin{algorithmic}[1]
+    \Ensure Zero Based Indexing
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: Use one Based Indexing for both string and matrix. Makes the code clutter-free.

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    \Statex
+    
+    \Function{Longest\_Palindrome}{$str$}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: Use Proper function name. Moreover, the function name is printed in all caps. To overcome this, use Math Mode

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    \LeftComment If $dp[i][j]$ is \textit{true}, $str[i...j]$ is a palindrome.
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: Just reworded it. (See Gist)

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 5]
```diff
@@ -0,0 +1,72 @@
+% Set the Page Layout
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex:1]: See the gist file that I've shared at the bottom to compare your changes

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+  \caption{Find the longest palindromic sequence in a given string}
+  
+  \begin{algorithmic}[1]
+    \Ensure Zero Based Indexing
+    \Statex
+    
+    \Function{Longest\_Palindrome}{$str$}
+    \LeftComment If $dp[i][j]$ is \textit{true}, $str[i...j]$ is a palindrome.
+        \If{$str$ is empty}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: You can use `Ensure`. See gist.

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            \State \Return{$""$}
+        \EndIf
+        \State $size \gets str.len$
+        \State $dp \gets \{false\}$  
+        \For{$i = size-1 : 0$}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: You can use some blank lines using `\Statex`.

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            \EndFor
+        \EndFor
+        
+        \State $start \gets 0$
+        \State $max\_length \gets 1$
+        
+        \For{$i = 0 : size-1$}
+            \For{$j = i : size-1$}
+                \If{$dp[i][j] == false$}
+                    \State \textit{continue}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: Just wrote it in a clean way. (See gist)

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            \EndFor
+        \EndFor
+        \State \Return{$str.substring(start, max\_length)$}
+    \EndFunction
+  \end{algorithmic}
+  
+\end{algorithm}
+
+\end{document}
+
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex:86]: I've also included some macros to be printed in boldface. See gist for their usage.

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                \EndIf
+            \EndFor
+        \EndFor
+        
+        \State $start \gets 0$
+        \State $max\_length \gets 1$
+        
+        \For{$i = 0 : size-1$}
+            \For{$j = i : size-1$}
+                \If{$dp[i][j] == false$}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: Replaced `==` with `is`

**Review by ghost — CHANGES_REQUESTED** [review]
Gist [Link](https://gist.github.com/Just-A-Visitor/8f5e69d6511e520a2c87071daf8c3680)

**Comment by ghost** [comment]
**Next Steps**

1) Go to this [link](https://gist.github.com/Just-A-Visitor/8f5e69d6511e520a2c87071daf8c3680) and see the changes that I've made to your code. I've also written them inline here. Study those changes to know the conventions that I follow.

2) After you are satisfied, click on the resolve conversation in each comment.

3) Replace your `SourceCode.tex` with my gist.

4) Ping me and I'll merge your PR

**Comment by ghost** [comment]
Fixes https://github.com/Just-A-Visitor/Algorithmic-Pseudocode/issues/35

File: Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex (hunk 3 above)
Comment (yildirimyigit) [reply Pseudocode/Dynamic Programming/Longest Palindromic Substring/SourceCode.tex]: I didn't know about the effect of not using Math Mode in function name. I'll do that from now on.

**Comment by yildirimyigit** [comment]
So I just pushed the changes you suggested.

**Comment by ghost** [comment]
Great. I'll merge this now. Can you leave the link of your cpp code so that when I create the `Validation Codes` folder, I can add it there. 

Thanks for contributing to this repo!

**Comment by ghost** [comment]
Leaving this link here for reference

[Validation Code](http://cpp.sh/9bmyy)

**Comment by yildirimyigit** [comment]
I just created a gist, in case cpp.sh removes the code in the future. Here it is:

https://gist.github.com/yildirimyigit/f2e743f61f507c0927e8221d4c8e032b

**Pull request merged**